_posts/2015-03-stitch-fix-heart-unix.md

```diff
@@ -0,0 +1,139 @@
+*Proposed Outline:*
+```
+I. Introduction
+II. Table of contents by problem
+III. Each problem
+  A. Title
+    1. Attribution and environment sub-heading
+  B. Problem
+  C. Solution
+    1. Input
+    2. Command
+    3. Explanation
+IV. ~~Closing~~?
+```
+
+--
+
+I &#10084; UNIX and using the command line; they help me solve problems at Stitch Fix.  I'm not alone.  Across the Data Science and Engineering teams, we're constantly solving problems with UNIX and the command line.  **TODO: Closing sentence replaces closing section?**  
+
+   
+## Problems and Solutions
+ - Files
+   - [What size are these images?](#check-image-size)
+   - [What files recently changed?]()
+   - [How many lines are in this data dump?]()
+   - [How do you bulk change filenames?]()
+   - [Can I efficiently and securely copy these files?]()
+ - Directories
+   - [Directory File Summary]()
+   - [Sort Directories By Number Of Files]()
+   - [Delete Directory With Large Number Of Files]()
+ - CSVs
+   - [Filter CSV File By Column Values]()
+   - [Index CSV File]()
+ - Systems 
+   - [Display Resource Usage and Availability]()
+   - [Easily Manage Python Virtual Environments]()
+ - Productivity
+   - [Alias Everything]()
+   - [Command Completion Alert]()
+ - [Bemuse Coworkers]()
+
+<a name="check-image-size"></a>
+### Check Image Size
+&#8212; [Dave Copeland](https://twitter.com/davetron5000/) *(OS X / bash)*
+
+#### Problem
+Check the sizes of our product images the pickers use to check what they are picking in our warehouse
+
+#### Solution
+Created a script to run curl on the argument to download the image, use ImageMagick to get its size and print that out in CSV.  Piped my input CSV of images into `xargs -n1 -P8 ./my_script.rb` to basically run my script 8-way parallel to get the job done as fast as I could without setting my machine on fire.
+
+##### Input
+Simple CSV of some ids and an image URL
+
+##### Command
+```
+cat my_input.csv | xargs -n1 -P8 ./check.rb > images_with_possible_issues.csv
+```
+
+ - `xargs` runs a command, feeding `xargs` `STDIN` as arguments to that command
+   - `echo ""foo"" | xargs ls` is the same as `""ls foo""`.  
+ - `-n1` says to run the given command once for each line of input.  
+  - Normally `xargs` will run many lines at once, so if you had a file with 10 rows in it called `""blah.csv""` and do `cat blah.csv | xargs curl`, it would likely run curl once with all 10 rows of blah.csv given to curl.  `-n1` means to just do them one at a time (so 10 runs of curl in this example).
+ - `-P8` says to parallelize it 8 ways.  
+
+So, I've got 8 instances of my script running at once.  Obviously, there are diminishing returns on parallelism, but since curl'ing images is mostly I/O bound this worked pretty well without compromising my machine.
+
+### Find Recently Changed Files
```

Comment:
Dumb question, but in the final article each of these sections would be fleshed out?